src/Strategy/Conda.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Strategy.Conda (
+  discover,
+) where
+
+import Discovery.Walk
+import Effect.Exec
+import Effect.ReadFS
+import Control.Carrier.Diagnostics hiding (fromMaybe)
+import Path
+import Types
+import Graphing (Graphing)
+import qualified Strategy.Conda.CondaList as CondaList
+import qualified Strategy.Conda.EnvironmentYml as EnvironmentYml
+
+discover :: (Has ReadFS sig m, Has Diagnostics sig m, Has ReadFS rsig run, Has Exec rsig run, Has Diagnostics rsig run) => Path Abs Dir -> m [DiscoveredProject run]
+discover dir = map mkProject <$> findProjects dir
+
+findProjects :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs Dir -> m [CondaProject]
+findProjects = walk' $ \dir _ files -> do
+  case findFileNamed "environment.yml" files of
+    Nothing -> pure([], WalkContinue)
+    Just envYml -> do
+      let project =
+            CondaProject
+              {
+                condaDir = dir,
+                condaEnvironmentYml = envYml
+              }
+      pure([project], WalkSkipAll) -- Once we find an environment.yml file, skip the rest of the walk
+
+data CondaProject =
+    CondaProject
+      { condaDir :: Path Abs Dir,
+        condaEnvironmentYml :: Path Abs File
+      } deriving (Eq, Ord, Show)
+
+mkProject :: (Has ReadFS sig n, Has Exec sig n, Has Diagnostics sig n) => CondaProject -> DiscoveredProject n
+mkProject project =
+  DiscoveredProject
+    { projectType = "conda",
+      projectBuildTargets = mempty,
+      projectDependencyGraph = const $ getDeps project, -- what does const do here?
+      projectPath = condaDir project,
+      projectLicenses = pure []
+    }
+
+getDeps :: (Has Exec sig m, Has ReadFS sig m, Has Diagnostics sig m) => CondaProject -> m (Graphing Dependency)
+getDeps project = analyzeCondaList project <||> analyzeEnvironmentYml project
```

Comment:
We should use `combineSuccessful` here, so that we try both concurrently (or at least concurrent-able) and merge them if they are both successful.
```suggestion
getDeps project = combineSuccessful "failed to analyze conda project" [analyzeCondaList project,  analyzeEnvironmentYml project]
```

Reply:
Wouldn't that combine the results from both?

```
-- | Run a list of actions, combining the successful ones. If all actions fail, 'fatalText' is invoked with the provided @Text@ message.
```

Reply:
Is there a reason we shouldn't combine them?

Reply:
That doesn't really explain why we wouldn't combine them.  Any dependency found by both will be merge as part of a dedupe operation, so combining should be safe. The more expensive one will always be tried first, so there's no real performance concern.

Unless there's reason to expect that running both will produce incorrect results when combined, we should run both.

Reply:
Example:

There might be a dep with a version spec in an `environment.yml` file: i.e. `conda+foo$`1.2.*`, and perhaps the same dep resolved to a known version in the users virtual environment: i.e. `conda+foo$1.2.4` (we get that form `conda list`. If we combined the results then we would include both of those deps in the result, which is not correct behavior. We want to take `conda list` results if there, and if we can't get them for whatever reason, fallback and take the results from parsing the `environment.yml` file